app/lib/publisher/backend.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:gcloud/db.dart';
+import 'package:gcloud/service_scope.dart' as ss;
+import 'package:logging/logging.dart';
+import 'package:pub_server/repository.dart' show UnauthorizedAccessException;
+
+import '../account/backend.dart';
+
+import 'models.dart';
+
+final _logger = Logger('pub.publisher.backend');
+
+/// Sets the publisher backend service.
+void registerPublisherBackend(PublisherBackend backend) =>
+    ss.register(#_publisherBackend, backend);
+
+/// The active publisher backend service.
+PublisherBackend get publisherBackend =>
+    ss.lookup(#_publisherBackend) as PublisherBackend;
+
+/// Represents the backend for the publisher handling and related utilities.
+class PublisherBackend {
+  final DatastoreDB _db;
+
+  PublisherBackend(this._db);
+
+  /// Loads a publisher (or returns null if it does not exists).
+  Future<Publisher> getPublisher(String publisherId) async {
+    final pKey = _db.emptyKey.append(Publisher, id: publisherId);
+    return (await _db.lookup<Publisher>([pKey])).single;
+  }
+
+  /// Checks whether the current authenticated user has admin role of the
+  /// publisher, and executes [fn] if it does.
+  Future<R> _withPublisherAdmin<R>(
+      String publisherId, Future<R> fn(Publisher p)) async {
+    return await withAuthenticatedUser((user) async {
+      final p = await getPublisher(publisherId);
+      if (p == null) {
+        throw Exception('Publisher does not exists.');
+      }
+
+      final member = (await _db.lookup<PublisherMember>(
+              [p.key.append(PublisherMember, id: user.userId)]))
+          .single;
+      if (member == null || member.role != PublisherMemberRole.admin) {
+        _logger.info(
+            'Unauthorized access of Publisher($publisherId) from ${user.email}.');
+        throw UnauthorizedAccessException('User is not an admin.');
+      }
+      return await fn(p);
+    });
+  }
+
+  /// Updates the publisher data.
+  Future updatePublisherData(String publisherId, String description) async {
+    if (description == null) {
+      throw Exception('Description must not be null.');
```

Comment:
Is this an `Exception` or an `Error`, seems like an `ArgumentError`, consider:
```dart
ArgumentError.checkNotNull(description, 'description');
```
See: https://api.dartlang.org/stable/2.4.0/dart-core/ArgumentError/checkNotNull.html

We should do this whenever it's wrong to call a method with `null` as an argument.

-----
Side note, I suspect it's almost always wrong to throw `Exception`, as users of the function should never be doing a catch-all, thus, we should throw a subclass, either a custom or existing one. If you're intending to communicate this to the request handler and further up to the API caller, then it's probably best to throw a custom `Exception` and catch it later.

We should never catch `Exception` and then print the `e.toString()` in a `400` response to the user. Because we could be exposing exceptions that we didn't intend to catch. We could catch an exception from a grpc/http client that embeds the authentication token used or request signature in the exception message (unlikely, but be defensive). If we get an exception we didn't anticipate it's always an internal error -> 500.